### PR TITLE
Merge makefile for e2e perf test

### PR DESCRIPTION
**Pull request by qiuming-best**
Thank you for contributing to Velero!

# Please add a summary of your change
Merging e2e test makefile and perf test makefile into one makefile. only modifying one makefile when Parameters change for both tests.
# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by blackpiglet — DISMISSED** [review]

**Review by sseago — DISMISSED** [review]

**Comment by sseago** [comment]
@qiuming-best @blackpiglet I approved this one, but I noticed that the linter seems to be failing. Looking at the output, I see a bunch of errors that shouldn't be errors -- any idea whether something is off with the configuration? Could the bump up to golang 1.22 have broken something?

**Review by kaovilai — APPROVED** [review]

File: test/Makefile [hunk 1]
```diff
@@ -176,6 +191,48 @@ run: ginkgo
...
+.PHONY: run-perf
+run-perf: ginkgo
+		@[ "${CREDS_FILE}" ] && echo "Using credentials from ${CREDS_FILE}" || \
+			( echo "A credentials file is required to run E2E tests, please re-run the make target with CREDS_FILE=<PathToCredentialsFile>"; exit 1 )
+		@[ "${BSL_BUCKET}" ] && echo "Using bucket ${BSL_BUCKET} to store backups from E2E tests" || \
+			(echo "Bucket to store the backups from E2E tests is required, please re-run with BSL_BUCKET=<BucketName>"; exit 1 )
+		@[ "${CLOUD_PROVIDER}" ] && echo "Using cloud provider ${CLOUD_PROVIDER}" || \
+			(echo "Cloud provider for target cloud/plugin provider is required, please rerun with CLOUD_PROVIDER=<aws,azure,kind,vsphere>"; exit 1)
+	@$(GINKGO) -v $(FOCUS_STR) $(SKIP_STR) ./perf -- -velerocli=$(VELERO_CLI) \
+		-velero-image=$(VELERO_IMAGE) \
```
Comment (danfengliu) [inline test/Makefile:203]: Consider to move out shared parameters with run-e2e, then make it reusable to run-perf.

**Review by danfengliu — CHANGES_REQUESTED** [review]

File: test/Makefile (hunk 1 above)
Comment (qiuming-best) [reply test/Makefile]: Adjusted

**Comment by qiuming-best** [comment]
> @qiuming-best @blackpiglet I approved this one, but I noticed that the linter seems to be failing. Looking at the output, I see a bunch of errors that shouldn't be errors -- any idea whether something is off with the configuration? Could the bump up to golang 1.22 have broken something?

@sseago This is because [PR](https://github.com/vmware-tanzu/velero/pull/7573) add some strict checks for test directory. And @blackpiglet will help to solve it.

**Comment by blackpiglet** [comment]
@sseago 
It should relate to the Golang version bump. 
This PR #7697 also contains the linter fix code.

**Review by danfengliu — APPROVED** [review]
LGTM

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**